api/filter/http_connection_manager.proto

```diff
@@ -148,6 +148,14 @@ message AccessLog {
 
   // Filter which is used to determine if the access log needs to be written.
   AccessLogFilter filter = 3;
+
+  // The name of the access log implementation to instantiate. The name must match
+  // a supported access log.
+  string name = 6;
```

Comment:
In looking at this, I'm a little confused about how text based format logging interacts with RPC logging. Is the format string basically ignored in this case? The reason I ask is that we are likely going to add proto3 logging format directly to OSS code at some point for Lyft use (I imagine others will want this as well). Just trying to understand how that fits in.

Reply:
Yes, that is my understanding.  We would like to have more general pluggability here on top of the OSS proto3 RPC logging for other use cases.  Ignoring these fields is essentially a compromise because the api is frozen.  If we were to be able to unfreeze it, it would probably make more sense to have much of the RPC-specific or text-specific options be put into opaque configurations (leaving the `AccessLogFilter` objects outside because those are pretty universal) specific to their implementations.  I just realized that we haven't really discussed statically registered access log implementations before as we have the other statically registered components, so maybe we should open an issue on this since it deviates from the previous v2 api direction?

Reply:
AFAIK no one is yet using this part of the API in production, so I think if we wanted to do some targeted fixes here, we could do it now (if we do it relatively quickly). My vote would actually be to clean this up a little bit to account for the different use cases: 1) traditional text based format logging, 2) future proto3 logging, etc., 3) custom site specific logging.

@htuch thoughts?

Reply:
I just looked through the protos again, and I guess what we have now is OK since we do specify the access log proto and talk about RPC. I guess I'm a little confused about how the static stuff is going to fit in if still using the proto3 definitions and additional headers, etc. Maybe more comments?

Reply:
If we're good with briefly thawing this bit of the API (which it seems is the case), I'd +1 @mrice32 idea of just making it fully pluggable (while providing the text based version in a message in `http_connection_manager.proto`).

Reply:
+1

Reply:
Great!  I'll update.